src/Model/Resolver/Products/DataPostProcessor/PriceRange.php

```diff
@@ -0,0 +1,435 @@
+            ];
+        }
+
+        // Bundle products have special price set in % (percents)
+        $specialPricePrecentage = $this->getSpecialProductPrice($product);
+        $percentOff = is_null($specialPricePrecentage) ? 0 : 100 - $specialPricePrecentage;
+
+        return [
+            'amount_off' => $regularPrice * ($percentOff / 100),
+            'percent_off' => $percentOff
+        ];
+    }
+
+    /**
+     * Get value difference between two prices
+     *
+     * @param float $regularPrice
+     * @param float $finalPrice
+     * @return float
+     */
+    protected function getPriceDifferenceAsValue(float $regularPrice, float $finalPrice)
+    {
+        $difference = $regularPrice - $finalPrice;
+
+        if ($difference <= $this->zeroThreshold) {
+            return 0;
+        }
+
+        return round($difference, 2);
+    }
+
+    /**
+     * Get percent difference between two prices
+     *
+     * @param float $regularPrice
+     * @param float $finalPrice
+     * @return float
+     */
+    protected function getPriceDifferenceAsPercent(float $regularPrice, float $finalPrice)
+    {
+        $difference = $this->getPriceDifferenceAsValue($regularPrice, $finalPrice);
+
+        if ($difference <= $this->zeroThreshold || $regularPrice <= $this->zeroThreshold) {
+            return 0;
+        }
+
+        return round(($difference / $regularPrice) * 100, 8);
+    }
+
+    /**
+     * Gets [active] special price value
+     *
+     * @param Product $product
+     * @return float
+     */
+    protected function getSpecialProductPrice(Product $product): ?float
+    {
+        $specialPrice = $product->getSpecialPrice();
+        if (!$specialPrice) {
+            return null;
+        }
+
+        // Special price range
+        $from = strtotime($product->getSpecialFromDate());
+        $to = $product->getSpecialToDate() === null ? null : strtotime($product->getSpecialToDate());
+        $now = time();
+
+        return ($now >= $from && $now <= $to) || ($now >= $from && is_null($to)) ? (float)$specialPrice : null;
+    }
+
+    protected function isPriceIncludesTax(){
+        return $this->scopeConfig->getValue(
+            self::XML_PRICE_INCLUDES_TAX,
+            ScopeInterface::SCOPE_STORES
+        );
+    }
+
+    /**
+     * @inheritDoc
+     * @throws Exception
+     */
+    public function process(
+        array $products,
+        string $graphqlResolvePath,
+        $graphqlResolveInfo,
+        ?array $processorOptions = []
+    ): callable {
+        $productPrices = [];
+
+        $fields = $this->getFieldsFromProductInfo(
+            $graphqlResolveInfo,
+            $graphqlResolvePath
+        );
+
+        if (!count($fields)) {
+            return function (&$productData) {
+            };
+        }
+
+        $store = $this->storeManager->getStore();
+
+        /** @var Product $product */
+        foreach ($products as $product) {
```

Comment:
okay, I'm not seeing it
is there much actual benefit on it being a post processor, if you're just executing all the old code for each product, one at a time?
is there any grouped data requests as compared to what the original was?

Reply:
For now removed it from this PR. Wanna to split it, because not really sure how much time it will take to make it.  I guess QA can test right now options just.